lib/commands/push.ts

```diff
@@ -234,146 +258,165 @@ export default class PushCmd extends Command {
 			PushCmd,
 		);
 
+		const logger = await Command.getLogger();
+		logger.logDebug(`Using build source directory: ${options.source} `);
+
 		const sdk = getBalenaSdk();
 		const { validateProjectDirectory } = await import('../utils/compose_ts');
-
-		const source = options.source || '.';
-		if (process.env.DEBUG) {
-			console.error(`[debug] Using ${source} as build source`);
-		}
-
 		const { dockerfilePath, registrySecrets } = await validateProjectDirectory(
 			sdk,
 			{
 				dockerfilePath: options.dockerfile,
-				noParentCheck: options['noparent-check'] || false,
-				projectPath: source,
+				noParentCheck: options['noparent-check'],
+				projectPath: options.source,
 				registrySecretsPath: options['registry-secrets'],
 			},
 		);
 
-		const nogitignore = !options.gitignore;
-		const convertEol = !options['noconvert-eol'];
-
-		const appOrDevice = params.applicationOrDevice;
-		const buildTarget = await this.getBuildTarget(appOrDevice);
-		switch (buildTarget) {
+		switch (await this.getBuildTarget(params.applicationOrDevice)) {
 			case BuildTarget.Cloud:
-				const remote = await import('../utils/remote-build');
-
-				// Check for invalid options
-				const localOnlyOptions = ['nolive', 'service', 'system', 'env'];
-
-				localOnlyOptions.forEach((opt) => {
-					// @ts-ignore : Not sure why typescript wont let me do this?
-					if (options[opt]) {
-						throw new ExpectedError(
-							`The --${opt} flag is only valid when pushing to a local mode device`,
-						);
-					}
-				});
-
-				const app = appOrDevice;
-				await Command.checkLoggedIn();
-				const [token, baseUrl, owner] = await Promise.all([
-					sdk.auth.getToken(),
-					sdk.settings.get('balenaUrl'),
-					this.getAppOwner(sdk, app),
-				]);
-
-				const opts = {
+				logger.logDebug(
+					`Attempting remote build for application: ${params.applicationOrDevice}`,
```

Comment:
This a new log message, right? I'd suggest a more optimistic tone than "attempting", :-) say "starting":
```
`Starting cloud build for application: ${params.applicationOrDevice}`
```
```
`Starting local build on device: ${params.applicationOrDevice}`
```